Add README file for Helm charts section (#338)
istio-ecosystem/sail-operator: chart/README.md

    NAME: sail-operator
    LAST DEPLOYED: Mon Sep 16 12:43:18 2024
    NAMESPACE: sail-operator
    STATUS: deployed
    REVISION: 1
    TEST SUITE: None
    ```

5. Check `sail-operator` deployment is successfully installed and its pods are running:

    ```sh
    $ kubectl -n sail-operator get deployment --output wide

    NAME            READY   UP-TO-DATE   AVAILABLE   AGE   CONTAINERS                IMAGES                                                                                    SELECTOR
    sail-operator   1/1     1            1           19m   kube-rbac-proxy,manager   gcr.io/kubebuilder/kube-rbac-proxy:v0.16.0,quay.io/maistra-dev/sail-operator:0.1.0-rc.1   app.kubernetes.io/created-by=sailoperator,app.kubernetes.io/part-of=sailoperator,control-plane=sail-operator

    $ kubectl -n sail-operator get pods -o wide

    NAME                             READY   STATUS    RESTARTS   AGE   IP           NODE                 NOMINATED NODE   READINESS GATES
    sail-operator-666f84b6f4-9hw4t   2/2     Running   0          43s   10.244.0.8   sail-control-plane   <none>           <none>
    ```

## Deploying Istio

To deploy Istio, you must create the following resources:

> * In Kubernetes - `Istio`.
> * In OpenShift - `Istio` and `IstioCNI`.
> The `Istio` resource deploys and configures the Istio Control Plane, whereas the `IstioCNI` resource (in OpenShift) deploys and configures the Istio CNI plugin. You should create these resources in separate projects.

Reviewer: ```suggestion
* `Istio`.
* If you are using OpenShift, the `IstioCNI` must also be created.

The `Istio` resource deploys and configures the Istio Control Plane, whereas the `IstioCNI` resource (in OpenShift) deploys and configures the Istio CNI plugin. You should create these resources in separate projects.
```

Author: Modified.

### Create a namespace for Istio project.

* Kubernetes

    ```sh
    $ kubectl create namespace istio-system
    ```

* OpenShift

    ```sh
    $ kubectl create namespace istio-system
    $ kubectl create namespace istio-cni
    ```

### Create the Istio resource

The `sail-operator` charts directory contains `samples` directory, which contains manifests that could be used for Istio deployment.

* Kubernetes

    ```sh
    $ kubectl apply -f sail-operator/samples/istio-sample-kubernetes.yaml
    ```

* OpenShift

    ```sh
    $ kubectl apply -f sail-operator/samples/istio-sample-openshift.yaml
    $ kubectl apply -f sail-operator/samples/istiocni-sample.yaml
    ```

> **Note** - The Istio version deployment could be specified by modifying the `version` field within `Istio` and `IstioCNI` manifests.

Reviewer: ```suggestion
**Note** - The version can be specified by modifying the `version` field within `Istio` and `IstioCNI` manifests.
```

# Deploy Sail-Operator by using Helm charts

Follow this guide to install and configure Sail Operator by using [Helm](https://helm.sh/docs/)

## Prerequisites

Kubernetes:
* You have deployed a cluster on Kubernetes platform 1.27 or later.
* You are logged in to the Kubernetes cluster with admin permissions level user.

OpenShift:

> * You have deployed a cluster on OpenShift Container Platform 4.13 or later.  

Author: Fixed

* Extract it locally.

    ```sh
    $ tar -xvf /tmp/sail-operator-<release-number>.tgz
    ```

The extract command will create the `sail-operator` directory with the helm charts in it.

## Installation steps

This section describes the procedure to install `Sail Operator` using Helm. The general syntax for helm installation is:

    ```sh
    helm install <release> <chart> --create-namespace --namespace <namespace> [--set <other_parameters>]
    ```

The variable specified in the command are as follows:
* `<release>` - A name to identify and manage the Helm chart once installed.
* `<chart>` - A path to a packaged chart, a path to an unpacked chart directory or a URL.
* `<namespace>` - The namespace in which the chart is to be installed.

Default configuration values can be changed using one or more `--set <parameter>=<value>` arguments. Alternatively, you can specify several parameters in a custom values file using the `--values <file>` argument.

1. Create the namespace, `sail-operator`, for the Sail Operator components:

    ```sh
    $ kubectl create namespace sail-operator
    ```

**Note** - This step could be skipped of using the `--create-namespace` argument in step 2.

> 2. Install the Sail Operator base charts which contains cluster-wide Custom Resource Definitions (CRDs) which must be installed prior to the deployment of the Istio control plane:

Author: Modified.

* Kubernetes

    ```sh
    $ helm install sail-operator sail-operator/ --namespace sail-operator
    ```

* OpenShift

    ```sh
    $ helm install sail-operator sail-operator/ --namespace sail-operator --set platform=openshift
    ```

3. Validate the CRD installation with the `helm ls` command:

    ```sh
    $ helm ls -n sail-operator

    NAME         	NAMESPACE    	REVISION	UPDATED                                	STATUS  	CHART                   	APP VERSION
    sail-operator	sail-operator	1       	2024-09-16 12:43:18.786846217 +0300 IDT	deployed	sail-operator-0.1.0-rc.1	0.1.0-rc.1
    ```

4. Get the status of the installed helm chart to ensure it is deployed:

    ```bash
    $ helm status sail-operator -n sail-operator

    NAME: sail-operator
    LAST DEPLOYED: Mon Sep 16 12:43:18 2024
    NAMESPACE: sail-operator
    STATUS: deployed
    REVISION: 1
    TEST SUITE: None
    ```

> 5. Check `sail-operator` service is successfully installed and its pods are running:

Author: Changed and added.

> # Deploy Sail-Operator by using Helm charts

Reviewer: JFYI, we are using the convention Sail Operator in our docs.

```suggestion
# Deploy Sail Operator by using Helm charts
```

Author: Fixed

## Prerequisites

Kubernetes:
* You have deployed a cluster on Kubernetes platform 1.27 or later.
* You are logged in to the Kubernetes cluster with admin permissions level user.

OpenShift:
* You have deployed a cluster on OpenShift Container Platform 4.14 or later.  
* You are logged in to the OpenShift Container Platform web console as a user with the `cluster-admin` role.

[Install the Helm client](https://helm.sh/docs/intro/install/), version 3.6 or above.

## Prepare the Helm charts

**Note** - `Sail Operator` could be installed by downloading the release artifacts from the [release page](https://github.com/istio-ecosystem/sail-operator/releases).

* Download the required release artifact
* Extract it locally.

    ```sh
    $ tar -xvf /tmp/sail-operator-<release-number>.tgz
    ```

The extract command will create the `sail-operator` directory with the helm charts in it.

## Installation steps

This section describes the procedure to install `Sail Operator` using Helm. The general syntax for helm installation is:

    ```sh
    helm install <release> <chart> --create-namespace --namespace <namespace> [--set <other_parameters>]
    ```

> The variable specified in the command are as follows:

Reviewer: ```suggestion
The variables specified in the command are as follows:
```

Author: Fixed

* `<release>` - A name to identify and manage the Helm chart once installed.
* `<chart>` - A path to a packaged chart, a path to an unpacked chart directory or a URL.
* `<namespace>` - The namespace in which the chart is to be installed.

Default configuration values can be changed using one or more `--set <parameter>=<value>` arguments. Alternatively, you can specify several parameters in a custom values file using the `--values <file>` argument.

1. Create the namespace, `sail-operator`, for the Sail Operator components:

    ```sh
    $ kubectl create namespace sail-operator
    ```

> **Note** - This step could be skipped of using the `--create-namespace` argument in step 2.

Reviewer: ```suggestion
**Note** - This step could be skipped by using the `--create-namespace` argument in step 2.
```

    kind: Istio
    metadata:
    name: example
    spec:
    version: v1.23.0
    values:
        global:
        mtls:
            enabled: true
        trustDomainAliases:
        - example.net
        meshConfig:
        trustDomain: example.com
        trustDomainAliases:
        - example.net
    ```

For a list of available configuration for the `spec.values` field, run the following command:

    ```sh
    $ kubectl explain istio.spec.values
    ```

For the `IstioCNI` resource, replace `istio` with `istiocni` in the command above.

Alternatively, refer to [Istio's artifacthub chart documentation](https://artifacthub.io/packages/search?org=istio&sort=relevance&page=1) for:

- [Base parameters](https://artifacthub.io/packages/helm/istio-official/base?modal=values)
- [Istiod parameters](https://artifacthub.io/packages/helm/istio-official/istiod?modal=values)
- [Gateway parameters](https://artifacthub.io/packages/helm/istio-official/gateway?modal=values)
- [CNI parameters](https://artifacthub.io/packages/helm/istio-official/cni?modal=values)
- [ZTunnel parameters](https://artifacthub.io/packages/helm/istio-official/ztunnel?modal=values)

> ## Installing the istioctl tool

Reviewer: Since this content is similar to https://github.com/istio-ecosystem/sail-operator/tree/main/bundle#installing-the-istioctl-tool, it would be good if we avoid the duplication by pointing to the appropriate page (or re-arranging the content to its own page and linking them where necessary).

Author: That's a good point.

I made some re-arragement of the docs to avoid such duplication.
